crates/core/src/models/mod.rs

```diff
@@ -234,3 +239,73 @@ impl ConvertibleStarknetBlock for BlockWithTxs {
         Ok(block.into())
     }
 }
+
+pub struct StarknetEvent(Event);
+
+impl StarknetEvent {
+    pub fn new(sn_event: Event) -> Self {
+        Self(sn_event)
+    }
+}
+
+#[async_trait]
+impl ConvertibleStarknetEvent for StarknetEvent {
+    async fn to_eth_log(&self, _client: &dyn KakarotClient) -> Result<Log, KakarotClientError> {
+        // let starknet_block_id = StarknetBlockId::Tag(BlockTag::Latest);
+        // let address = match client.get_evm_address(&self.0.from_address, &starknet_block_id).await {
+        //     Ok(address) => address,
+        //     Err(_) => {
+        //         return Err(KakarotClientError::OtherError(anyhow::anyhow!(
+        //             "Starknet event address has no EVM mapping",
+        //         )));
+        //     }
+        // };
+
+        // Convert the StarkNet address to a byte array
+        let sn_bytes = self.0.from_address.clone().to_bytes_be();
+        // Take the last 20 bytes of the StarkNet address to create the Ethereum address
+        let address = Address::from_slice(&sn_bytes[12..32]);
+
+        let topics: Result<Vec<H256>, KakarotClientError> = self
+            .0
+            .keys
+            .chunks(2)
+            .map(|chunk| {
+                let low = BigUint::from_bytes_be(&chunk[0].to_bytes_be());
+                let high = match chunk.get(1) {
+                    Some(h) => BigUint::from_bytes_be(&h.to_bytes_be()),
+                    None => {
+                        return Err(KakarotClientError::OtherError(anyhow::anyhow!(
+                            "Not an convertible event: High value doesn't exist",
+                        )));
+                    }
+                };
+                let result = low + (BigUint::from(2u128).pow(128u32) * high);
+                // Converts the result to bytes.
+                let bytes = result.to_bytes_be();
+                // If the length of bytes is less than 32, prepends it with zeros to make it 32 bytes long.
+                let bytes = once(0u8).cycle().take(32 - bytes.len()).chain(bytes.into_iter()).collect::<Vec<_>>();
+                Ok(H256::from_slice(&bytes))
+            })
+            .collect();
+
+        let topics = match topics {
```

Comment:
you can add a "?" at the end of the topics definition, i.e. after `collect()`, 
if the error cannot be implicitly converted, then you can add `.map_err`